pkg/operator/applier/kubectl.go

```diff
@@ -132,6 +136,138 @@ commonAnnotations:
 	return stdout, stderr, errors.Wrap(err, "failed to run kubectl apply")
 }
 
+// ApplyCreateOrPatch attempts to run a `kubectl apply` on the yaml document. If it fails
+func (c *Kubectl) ApplyCreateOrPatch(targetNamespace string, slug string, yamlDoc []byte, dryRun bool, wait bool, annotateSlug bool) ([]byte, []byte, error) {
+
+	stdout, stderr, err := c.Apply(targetNamespace, slug, yamlDoc, dryRun, wait, annotateSlug)
+	if !strings.Contains(string(stderr), "metadata.annotations: Too long") || err == nil {
+		return stdout, stderr, errors.Wrap(err, "failed attempted kubectl apply")
+	}
+	log.Println("Failed to apply document: metadata too long. Trying to 'create' or 'patch' instead")
+
+	docs, err := splitter.SplitYAML(yamlDoc)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "split yaml")
+	}
+
+	var combinedStdout, combinedStderr []byte
+	for name, b := range docs {
+		stdout, stderr, err = c.createOrPatchSingleDoc(name, targetNamespace, slug, b, dryRun, wait, annotateSlug)
+		combinedStdout = append(combinedStdout, stdout...)
+		combinedStderr = append(combinedStderr, stderr...)
+
+		if err != nil {
+			return combinedStdout, combinedStderr, errors.Wrap(err, "failed to run kubectl create")
+		}
+	}
+
+	return combinedStdout, combinedStderr, nil
+}
+
+func (c *Kubectl) createOrPatchSingleDoc(name string, targetNamespace string, slug string, yamlDoc []byte, dryRun bool, wait bool, annotateSlug bool) ([]byte, []byte, error) {
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to create temp directory")
+	}
+	defer os.Remove(tmp)
```

Comment:
RemoveAll is for directories
```suggestion
	defer os.RemoveAll(tmp)
```